src/cart/crt_swim.c

```diff
@@ -244,10 +313,26 @@ static int crt_swim_send_message(struct swim_context *ctx, swim_id_t to,
 	crt_rpc_t		*rpc_req;
 	crt_endpoint_t		 ep;
 	crt_opcode_t		 opc;
+	swim_id_t		 self_id = swim_self_get(ctx);
 	int			 opc_idx = 0;
 	int			 ctx_idx = csm->csm_crt_ctx_idx;
 	int			 rc;
 
+	if (nupds > 0 && upds[0].smu_state.sms_status == SWIM_MEMBER_INACTIVE)
+		opc_idx = 1;
+
+	opc = CRT_PROTO_OPC(CRT_OPC_SWIM_PROTO, CRT_OPC_SWIM_VERSION, opc_idx);
+
+	if (CRT_SWIM_SHOULD_FAIL(d_fa_swim_drop_rpc, self_id)) {
+		rc = -d_fa_swim_drop_rpc->fa_err_code;
```

Comment:
fa_err_code comes from the config file here doesn't it?  Why are you taking it's negative value in which case?

Reply:
yes, this come from YAML config. fa_err_code is unit32_t. I have to return a negative error.

Reply:
That should be changed to int32_t or possibly just int then, the purpose of fa_erro_code is to put daos error numbers in.

Reply:
It will be not consistent anyway. We have definition of DER_* as positive codes. Those error codes returned as "-DER_*".

Reply:
But once they're variables they're always negative, i'd assume you'd put negative values in the config file as that's how they all arrear in log files/messages.

Reply:
It's readed by strtoul() now. So, if you specify it as negative in config it will be readed as zero.

Reply:
Then use strtol().

Reply:
I agree with Ashley here; we should change this to strtol() when parsing arg/int32 and return code as is from config file instead of negating it

Reply:
Can we do this in separate PR?

Reply:
Sure, please file a task jira ticket then on this so that this does not get forgotten

Reply:
DAOS-6242

Reply:
https://github.com/daos-stack/daos/pull/4041